### PR TITLE
docs

### PR DESCRIPTION
**Pull request by jordens**
- **introduce custom .split().skip(1) impl**
- **refine Transcode docs**

### CODE REVIEW

File: miniconf/src/node.rs [hunk 1]
```diff
@@ -173,10 +173,45 @@ impl<T, const S: char> From<T> for Path<T, S> {
...
+
+impl<'a, const S: char> Iterator for PathIter<'a, S> {
+    type Item = &'a str;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.map(|s| {
+            let pos = s
+                .chars()
+                .map_while(|c| (c != S).then_some(c.len_utf8()))
```
Comment (ryan-summers) [inline miniconf/src/node.rs:204]: Would https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.skip_while work for us? I.e. just skip the empty string?

Comment (ryan-summers) [reply miniconf/src/node.rs:204]: Ah I think the docs about about empty keys vs empty path for options above explains it.

Comment (jordens) [reply miniconf/src/node.rs:204]: I think skip while empty would work but it's less strict than the  implicit definition above (a path element is always prefixed by a separator). And we'd have to be much kore careful w.r.t. composability in e.g. CLI, Mqtt, and serial settings where then we'd have to conditionally insert the separator if we allow path to start without.

Comment (ryan-summers) [reply miniconf/src/node.rs:204]: Yeah being strict here feels more right for sure. Thanks for humoring me

File: miniconf/src/node.rs [hunk 2]
```diff
@@ -173,10 +173,47 @@ impl<T, const S: char> From<T> for Path<T, S> {
...
+                .sum();
+            let (left, right) = s.split_at(pos);
+            self.0 = right.get(S.len_utf8()..);
```
Comment (ryan-summers) [inline miniconf/src/node.rs:207]: How is this semantically different than `split(S).skip(1)`?

Comment (jordens) [reply miniconf/src/node.rs:207]: `PathIter` has exactly the same Iterator behavior as `Skip<Split<_>>`. It's just 2 kB less code on stabilizer.
https://github.com/quartiq/miniconf/blob/0cd1773c192d14ad5423595d7817fda9df161785/miniconf/src/node.rs#L306-L314

Comment (ryan-summers) [reply miniconf/src/node.rs:207]: Huh, interesting. Alright, seems fine to me.

**Review by ryan-summers — APPROVED** [review]

**Pull request merged**